bundle/src/main/java/com/jfrog/ide/eclipse/scan/ScanManager.java

```diff
@@ -1,110 +1,202 @@
-import com.jfrog.ide.common.filter.FilterManager;
-import com.jfrog.ide.common.log.ProgressIndicator;
-import com.jfrog.ide.common.scan.ComponentPrefix;
-import com.jfrog.ide.eclipse.configuration.XrayServerConfigImpl;
+import com.jfrog.ide.common.parse.SarifParser;
+import com.jfrog.ide.eclipse.configuration.CliDriverWrapper;
 import com.jfrog.ide.eclipse.log.Logger;
-import com.jfrog.ide.eclipse.log.ProgressIndicatorImpl;
 import com.jfrog.ide.eclipse.scheduling.CliJob;
-import com.jfrog.ide.eclipse.ui.FilterManagerSingleton;
+import com.jfrog.ide.eclipse.ui.issues.ComponentIssueDetails;
 import com.jfrog.ide.eclipse.ui.issues.IssuesTree;
-import com.jfrog.ide.eclipse.utils.ProjectsMap;
-import com.jfrog.xray.client.services.summary.Components;
 import org.jfrog.build.api.util.Log;
 
 /**
  * @author yahavi
  */
-public abstract class ScanManager {
-
+public class ScanManager {
 	static final Path HOME_PATH = Paths.get(System.getProperty("user.home"), ".jfrog-eclipse-plugin");
+	private static ScanManager instance;
 	private IProgressMonitor monitor;
-	IProject project;
-	Log log;
-	JfrogCliDriver cliDriver;
+	private IWorkspace iworkspace;
+	private IProject[] projects;
+	private Log log;
+	private JfrogCliDriver cliDriver;
+	private SarifParser sarifParser;
+	private AtomicBoolean scanInProgress = new AtomicBoolean(false);
 	
-	ScanManager(IProject project, ComponentPrefix prefix) throws IOException {
-		this.project = project;
-		Files.createDirectories(HOME_PATH);
-		log = Logger.getInstance();
+	private ScanManager() {
+		this.iworkspace = ResourcesPlugin.getWorkspace();
+		this.projects = iworkspace.getRoot().getProjects();
+		this.log = Logger.getInstance();
+		this.cliDriver = CliDriverWrapper.getInstance().getCliDriver();  
+		this.sarifParser = new SarifParser(log);
+	}
+	
+	public static ScanManager getInstance(){
+        if (instance == null) {
+            synchronized (ScanManager.class) {
+                if (instance == null) {
+                    instance = new ScanManager();
+                }
+            }
+        }
+        return instance;
 	}
-
-	/**
-	 * Collect and return {@link Components} to be scanned by JFrog Xray.
-	 * Implementation should be project type specific.
-	 * 
-	 * @return
-	 * 
-	 * @throws CoreException
-	 * @throws IOException
-	 * @throws JsonProcessingException
-	 */
 	
-	public IProject getIProject() {
-		return project;
+	public void startScan(Composite parent, boolean isDebugLogs) {
+		Map<String, String> auditEnvVars = new HashMap<>();
+		
+		// If scan is in progress - do not perform another scan
+		if (isScanInProgress()) {
+			log.info("Previous scan still running...");
+			return;
+		}
+		
+		scanInProgress.compareAndSet(false, true);
+		resetIssuesView(IssuesTree.getInstance());
+		ScanCache.getInstance().resetCache();
+		
+		// refresh projects list
+		projects = iworkspace.getRoot().getProjects();
+		if (projects.length == 0) {
+			log.info("No projects to scan.");
+		}
```

Comment:
```suggestion
		if (projects.length == 0) {
			log.info("No projects to scan.");
			scanFinished()
			return
		}
```
we can already return here, also shouldn't we report scan complete since we will not enter `scanAndUpdateResults` even once?

Reply:
I have changed the order of the code so er eill first check if there are projects to scan, then if a scan is in progress, if not - set atomic boolean to true, then reset the issues view an call `scanAndUpdateResults()` in a for loop for each project.